lib/CXGN/Trait.pm

```diff
@@ -239,14 +259,262 @@ has 'uri' => (isa => 'Str',
+	my $cvterm = $schema->resultset("Cv::Cvterm")->find(
+		{
+			name        => $cvterm_name,
+			cv_id       => $cv_id,
+			is_obsolete => 0
+		},
+		{ key => 'cvterm_c1' }
+	);
+
+	my $root_id = $cvterm->get_column('cvterm_id');
+
+
+
+	# check to see if specified ontology exists
+	my $db = $schema->resultset("General::Db")->find(
+		{
+			name => $ontology_name
+		},
+		{ key => 'db_c1' }
+	);
+
+	if (!defined($db)) {
+		return {error => "Ontology id does not exist"}
+	}
+
+	$ontology_id = $db->get_column('db_id');
+
+	# check to see if cvterm name already exists and don't attempt if so
+	my $cvterm_exists = $schema->resultset("Cv::Cvterm")->find(
+		{
+			name        => $name,
+			cv_id       => $cv_id,
+			is_obsolete => 0
+		},
+		{ key => 'cvterm_c1' }
+	);
+
+	if (defined($cvterm_exists)) {
+		return {error => "Variable with that name already exists"}
+	}
+
+	# lookup last numeric accession number in ontology if one exists so we can increment off that
+	my $q = "select accession from dbxref where db_id=".$ontology_id." and accession ~ ".q('^\d+$')." order by accession desc limit 1;";
+	my $sth = $self->bcs_schema->storage->dbh->prepare($q);
+	$sth->execute();
+	my ($accession) = $sth->fetchrow_array();
+
+	if (!defined($accession)) {
+		$accession = '0000001';
+	} else {
+		$accession++;
+	}
+
+	# get cvterm_id for VARIABLE_OF
+	my $variable_of_cvterm = $schema->resultset("Cv::Cvterm")->find(
+		{
+			name        => 'VARIABLE_OF',
+			cv_id       => $cv_id,
+			is_obsolete => 0
+		},
+		{ key => 'cvterm_c1' }
+	);
+
+	my $variable_of_id = $variable_of_cvterm->get_column('cvterm_id');
+
+	my $new_term;
+
+	# setup transaction for rollbacks in case of error
+	my $coderef = sub {
+
+		# add trait info to dbxref
+		my $new_term_dbxref = $schema->resultset("General::Dbxref")->create(
+			{ db_id       => $ontology_id,
+				accession => $accession,
+				version   => '1',
+			},
+			{ key => 'dbxref_c1' },
+		);
+
+		# add trait info to cvterm
+		$new_term = $schema->resultset("Cv::Cvterm")->create(
+			{ cv_id         => $cv_id,
+				name        => $name,
+				definition  => $description,
+				dbxref_id   => $new_term_dbxref->dbxref_id(),
+				is_obsolete => 0
+			});
+
+		# set cvtermrelationship VARIABLE_OF to put terms under ontology
+		# add cvterm_relationship entry linking term to ontology root
+		my $relationship = $schema->resultset("Cv::CvtermRelationship")->create(
+			{ type_id      => $variable_of_id,
+				subject_id => $new_term->get_column('cvterm_id'),
+				object_id  => $root_id
+			});
+
+		# add synonyms
+		foreach my $synonym (@{$synonyms}) {
+			$new_term->add_synonym($synonym);
+		}
+
+		# TODO: save scale properties
+		my $scale = CXGN::BrAPI::v2::Scales->new({
+			bcs_schema => $self->bcs_schema,
+			scale => $self->scale,
+			cvterm_id => $new_term->get_column('cvterm_id')
+		});
+
+		$scale->store();
+
+		if ($scale->{'error'}) {
+			return {error => $scale->{'error'}};
+		}
+
+		# TODO: save method properties
```

Comment:
Is this a todo anymore? Looks done in line 465

Reply:
Removed TODO from comment and pushed changes